source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,253 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <OgreVector3.h>
+#include <OgreCamera.h>
+
+#include "camera/CullingManager.h"
+#include "camera/CameraManager.h"
+#include "gamemap/GameMap.h"
+
+#include "entities/Creature.h"
+#include "utils/VectorInt64.h"
+#include "utils/LogManager.h"
+#include "render/RenderManager.h"
+#include <sstream>
+#include <algorithm>
+
+//! Values used to know whether to show and/or hide a mesh
+static const int HIDE =  1;
+static const int SHOW =  2;
+
+
+CullingManager::CullingManager(CameraManager* cameraManager):
+    mFirstIter(false),
+    mCm(cameraManager),
+    mCullTilesFlag(false)
+{
+    mActivePlanes =(Ogre::Plane(0, 0, 1, 0));
+    // init the Ogre vector
+    for (unsigned int i = 0; i < 8; ++i)
+    {
+        mOgreVectorsArray[i].x = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].y = static_cast<Ogre::Real>(0.0);
+        mOgreVectorsArray[i].z = static_cast<Ogre::Real>(0.0);
+    }
+}
+
+void CullingManager::cullTiles()
+{
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(VectorInt64(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+
+    OD_LOG_DBG( mOldWalk.debug());
+    OD_LOG_DBG( mWalk.debug());
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(SHOW | HIDE);
+}
+
+void CullingManager::startTileCulling()
+{
+    getIntersectionPoints();
+
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(VectorInt64(mOgreVectorsArray[ii]));
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+    mOldWalk = mWalk;
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+    hideAllTiles();
+    newBashAndSplashTiles(SHOW);
+
+    mCullTilesFlag = true;
+}
+
+
+void CullingManager::stopTileCulling()
+{
+    mCullTilesFlag = false;
+    mOldWalk = mWalk;
+    mWalk.mVertices.mMyArray.clear();
+    for (int ii = 0 ; ii < 8 ; ++ii)
+        mWalk.mVertices.mMyArray.push_back(VectorInt64(mOgreVectorsArray[ii]));
+
+    // create a slope -- a set of left and rigth path
+    mWalk.convex_hull();
+    mWalk.buildSlopes();
+
+    OD_LOG_DBG( mOldWalk.debug());
+    OD_LOG_DBG( mWalk.debug());        
+
+    // reset index pointers to the begging of collections
+    mOldWalk.prepareWalk();
+    mWalk.prepareWalk();
+
+    newBashAndSplashTiles(HIDE);
+    showAllTiles();
+
+}
+
+void CullingManager::hideAllTiles(void)
+{
+    GameMap* gm = mCm->getGameMap();
+    if (!gm)
```

Comment:
IMHO, there is no need to check for game map being null as it is already expected not to be null everywhere else. But no need to change if you prefer letting like that